torch/fx/experimental/Partitioner.py

```diff
@@ -156,6 +271,80 @@ def find_single_partition(self) -> None:
         partition_0.add_parent(self.root_partition)
         return
 
+    def size_based_partition(self, available_mem: Union[float, int]) -> None:
+        """This method partitions the graph based on memory size.
+           We assume all devices have the same memory size.
+           The basic idea is:
+           First, create a new partition.
+           Then travel the graph in the order of bfs level.
+           The travel only focuses on op nodes
+           (call_function, call_module, call_method)
+           Check if a op node and its input nodes
+           could fit into the current partition.
+           If the current partition is full, create a new one
+           and continue traveling nodes.
+           For orphan nodes (placeholder or constants are never used),
+           a new partition will create for each of them.
+           Then through self.combine_partition_based_on_size(),
+           partitions will be combined to keep
+           as less partitions as possible
+        """
+        bfs_info = _get_bfs_levels(self.graph_module)
+        bfs_level_list = bfs_info['bfs_level_list']
+        self.args_mapping = bfs_info['args_mapping']
+        self.users_mapping = bfs_info['users_mapping']
+        partition = self.create_partition()
+        # track the used mem for the current partition
+        used_mem = 0
+        for level in bfs_level_list:
+            for i in level:
```

Comment:
I don't understand what the purpose of the BFS levels is here. Wouldn't iterating over the nodes in the valid topological sort of the GraphModule do the same thing here?